spmd/compiler/fusion.py

```diff
@@ -162,22 +214,448 @@ def _scan_graph_for_fusion_elements(
     return element_list
 
 
-def run_comm_fusion(gm: fx.GraphModule) -> bool:
+def _copy_fe_to_buffer(
+    gi: GraphInfo, gm: fx.GraphModule, in_fe_list: list[FusionElement]
+) -> None:
+    """first half of fusion - move desired items to buffer and create graph"""
+    buffer_node = gi.global_buffer_node
+    buffer_size = gi.global_buffer_size
+
+    copy_list = in_fe_list
```

Comment:
Should we explicitly copy the list? What is tlist below doing?

Reply:
good  catch.  I actually just removed this as it was an artifact of when it was originally a single function (which was split into two). 
Now the incoming param is copy_list  instead of in_fe_list, and that is used continuously throughout the function.